### ISSUE TITLE
Allow for Application Initialization

### ISSUE THREAD

**Issue by wbaes** [spark-jobserver/spark-jobserver#288]
Feature Request:
Sometimes, it is nice to perform some length operation when the application is starting and right after the Spark Context has been created.
This is to avoid repetative overhead from each request being processed using the same context.
Please allow for this one-time initialization.
thanks

**Comment by tverbeiren** [comment]
@wbaes: I understand the need for initialization. 

I solved it by having a REST endpoint that only does this. When this job has run, the necessary RDDs are created and cached and the rest of the application is available. Does that make sense for you?

**Comment by wbaes** [comment]
Yes this "solution" is what we're doing now. TO me this is a workaround. The functionality is missing from JobServer
Also, if what you're creating is RDD's then its fine. If you created some broadcast objects then you would not be able to share those unless you created your own Custom context

**Comment by velvia** [comment]
The solution as I had pointed out is
1) A more general sharing mechanism for things other than RDDs
2) A way to run initial jobs when a context is created

There is someone interested in working on 1.
